modin/pandas/test/dataframe/test_window.py

```diff
@@ -246,20 +246,51 @@ def test_fillna_inplace():
 def test_frame_fillna_limit(data):
     pandas_df = pandas.DataFrame(data)
 
+    replace_pandas_series = pandas_df.columns.to_series()
+    replace_dict = replace_pandas_series.to_dict()
+    replace_pandas_df = pandas.DataFrame(
+        {col: pandas_df.index.to_series() for col in pandas_df.columns},
+        index=pandas_df.index,
+    )
+    replace_modin_series = pd.Series(replace_pandas_series)
+    replace_modin_df = pd.DataFrame(replace_pandas_df)
+
     index = pandas_df.index
 
     result = pandas_df[:2].reindex(index)
     modin_df = pd.DataFrame(result)
     df_equals(
         modin_df.fillna(method="pad", limit=2), result.fillna(method="pad", limit=2)
     )
+    df_equals(
+        modin_df.fillna(replace_dict, limit=2), result.fillna(replace_dict, limit=2)
+    )
+    df_equals(
+        modin_df.fillna(replace_modin_series, limit=2),
```

Comment:
We should probably iterate over different values of `limit`. I suggest adding this to parameters of this test. Also, it would be good to iterate over other parameters, which can detect failures by your opinion in the implementation. The same is true for Series test.

Reply:
Ok I added multiple values of `limit` in tests both for Series and DataFrame. Series test also had a missing vase of value=dict (which had a bug).